### ISSUE TITLE
Inference on AudioSet

### ISSUE THREAD

**Issue by nandacv** [kkoutini/PaSST#37]
Thank you for the code and inference script.
I understand that the PaSST model has been trained on AudioSet with sampling rate of 32kHz. 
I am trying to make inference using the pre trained model. 
Could you please let me know if I have to retrain the model with AudioSet (sampling rate of 16kHz) data to use it to make inference on 16kHz data  or is there any other way? 

Also, curious to know why did you use 32kHz instead of already available 16kHz AudioSet data?

Thanks in advance.

**Comment by nandacv** [comment]
Thank you for the reply. 
Should this be followed by model.train()  or I can start with inference. 
Following is the code can you confirm with which one I should be moving forward with

1. With model.train()

from hear21passt.base import get_basic_model,get_model_passt
import torch
# get the PaSST model wrapper, includes Melspectrogram and the default pre-trained transformer
model = get_basic_model(mode="logits")
print(model.mel) # Extracts mel spectrogram from raw waveforms.
print(model.net) # the transformer network.

from hear21passt.models.preprocess import AugmentMelSTFT

model.mel =  AugmentMelSTFT(n_mels=128, sr=16000, win_length=400, hopsize=160, n_fft=512, freqm=48,
                         timem=192,
                         htk=False, fmin=0.0, fmax=None, norm=1, fmin_aug_range=10,
                         fmax_aug_range=1000)

model.train()
# example inference
model.eval()
with torch.no_grad():
    # audio_wave has the shape of [batch, seconds*32000] sampling rate is 32k
    # example audio_wave of batch=3 and 10 seconds
    audio = torch.ones((3, 32000 * 10))*0.5
    audio_wave = audio.cuda()
    logits=model(audio_wave) 

2. Without model.train()

from hear21passt.base import get_basic_model,get_model_passt
import torch
# get the PaSST model wrapper, includes Melspectrogram and the default pre-trained transformer
model = get_basic_model(mode="logits")
print(model.mel) # Extracts mel spectrogram from raw waveforms.
print(model.net) # the transformer network.

from hear21passt.models.preprocess import AugmentMelSTFT

model.mel =  AugmentMelSTFT(n_mels=128, sr=16000, win_length=400, hopsize=160, n_fft=512, freqm=48,
                         timem=192,
                         htk=False, fmin=0.0, fmax=None, norm=1, fmin_aug_range=10,
                         fmax_aug_range=1000)

# example inference
model.eval()
with torch.no_grad():
    # audio_wave has the shape of [batch, seconds*32000] sampling rate is 32k
    # example audio_wave of batch=3 and 10 seconds
    audio = torch.ones((3, 32000 * 10))*0.5
    audio_wave = audio.cuda()
    logits=model(audio_wave) 


Also I assume, each of these logits should be followed by application of sigmoid function to get the output classes? Please correct me if I am wrong. 

Thanks in advance.

**Comment by nandacv** [comment]
Thank you for the reply. 
Can you please confirm if the following code looks good?

from hear21passt.base import get_basic_model,get_model_passt
import torch
#get the PaSST model wrapper, includes Melspectrogram and the default pre-trained transformer
model = get_basic_model(mode="logits")
print(model.mel) # Extracts mel spectrogram from raw waveforms.
from hear21passt.models.preprocess import AugmentMelSTFT
model.mel =  AugmentMelSTFT(n_mels=128, sr=16000, win_length=400, hopsize=160, n_fft=512, freqm=48,
                         timem=192,
                         htk=False, fmin=0.0, fmax=None, norm=1, fmin_aug_range=10,
                         fmax_aug_range=1000)
#example inference
model.eval()
with torch.no_grad():
    #audio_wave has the shape of [batch, seconds*16000] sampling rate is 16k
    #example audio_wave of batch=3 and 10 seconds
    audio = torch.ones((3, 16000 * 10))*0.5
    logits=model(audio)

Also I assume, these logits should be followed by application of sigmoid function to get the output classes? Please correct me if I am wrong.

Thanks in advance.

**Comment by kkoutini** [comment]
yes, this looks correct.
